ouroboros-network/src/Ouroboros/Network/BlockFetch/State.hs

```diff
@@ -70,11 +76,13 @@ fetchLogicIterations
   -> FetchDecisionPolicy header
   -> FetchTriggerVariables peer header m
   -> FetchNonTriggerVariables peer header block m
+  -> ConsensusFetchStateRefinementPreBlockFetch peer header block m
   -> m Void
 fetchLogicIterations decisionTracer clientStateTracer
                      fetchDecisionPolicy
                      fetchTriggerVariables
-                     fetchNonTriggerVariables =
+                     fetchNonTriggerVariables
+                     consensusRefinement =
```

Comment:
I wonder if this could have a more descriptive name, namely, what is refining and how.